### ISSUE TITLE
Price change notification

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/catalog-storefront#443]
Need to implement a mechanism which will notify Message Broker about price changes in Backoffcie

**AC:**
* Magento "Export API" sends a notification about price change via Message Bus
* Message contains required information about new price and current scope (website, customer group)
* Message should be sent only if the price was changed
* All prices changes should be honored
  * regular prices
  * special prices
  * tier price
  * custom option prices
  * the complex product option price (downloadable link, gift card amounts)
* Consumer accepts messages with price changes in the Message Broker 
  * No logic required, only accept message

**Tech details:**
* use mView based approach to react on price changes in the source tables
  * example of mView configuration <https://github.com/magento/saas-export/pull/117/files]
  * approach can be used after merging [https://github.com/magento/magento2/pull/30790]
  * mView should be used regardless on indexer modes (can be done in parallel in a separate task)
* The message format for price change:
  * product id list may be applied for mass price updates
  * topic name: "export.product.prices"
```java
[
  {
    meta: {
       type: "price_changed", 
       back_url: <ready-to-use URL to BackOffice to GetPrices API>,
    },
    data: [
        { 
         id: <product or option ids>,
         w:  <affected website>,
         cg: <affected customer group>,
       },
       ...
    ]
  
  },
   ...
]
{code}

* message format to send changes about tier prices updates (only with qty > 1, qty = 1 should be handled in previous case)
{code}
{
    meta: {
       type: "tier*price*changed", 
       // no back-url, data processed on Message Broker immediately 
    },
    data: [
      {
        id: <product id>,
        w: <affected website OR null for "all">,
        cg: <affected customer group OR null for "all">,
        qty: <qty>,
        price: <price>
      }, 
       ...
   ]

}  
{code}
* If tier price was deleted
{code}
{
    meta: {
       type: "price_deleted", 
       // no back-url, data processed on Message Broker immediately 
    },
    data: [
      {
        id: <product id>,
        w: <affected website OR null for "all">,
        cg: <affected customer group OR null for "all">,
        qty: <qty>, 
      }, 
       ...
   ]
}  
{code}
* If price  was deleted
{code}
{
    meta: {
       type: "price_deleted", 
       back_url: <ready-to-use URL to BackOffice to GetPrices API>,
    },
    data: [
      {
        id: [product ids],
        w: <affected website OR null for "all">,
        cg: <affected customer group OR null for "all">,
      }, 
       ...
   ]
}  
{code}
* If product was deleted. topic: "catalog.export.product.data"
{code}
{
    meta: {
       type: "product_deleted", 
       // no back-url, data processed on Message Broker immediately 
    },
    data: {ids: [affected ids]}
   >
}  
```

**Comment by aleksejsbaranovs10** [comment]
@magento I am working on this

**Comment by aleksejsbaranovs10** [comment]
1. Product regular price changed in global scope
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":null,
            "cg":null,
            "value":"12.000000"
         }
      }
   ]
}
```

2. Product special price changed in global scope
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "code":"special_price"
         },
         "data":{
            "id":"71",
            "w":null,
            "cg":null,
            "value":"10.000000"
         }
      }
   ]
}
```

3. Product regular price changed on product store view level (website scope of the attribute)
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":null,
            "value":"15.000000"
         }
      }
   ]
}
```

4. Product special price changed on product store view level (website scope of the attribute)
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "code":"special_price"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":null,
            "value":"12.000000"
         }
      }
   ]
}
```

5. Special price deleted in global scope
```
{
   "events":[
      {
         "meta":{
            "type":"price_deleted",
            "code":"special_price"
         },
         "data":{
            "id":"71",
            "w":null,
            "cg":null
         }
      }
   ]
}
```

6. Special price deleted on product store view level (website scope of the attribute)
```
{
   "events":[
      {
         "meta":{
            "type":"price_deleted",
            "code":"special_price"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":null
         }
      }
   ]
}
```

7. Customer group price added for all websites, all groups, qty = 1 and fixed price.
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "price_type":"fixed",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":null,
            "cg":null,
            "value":"8.000000"
         }
      }
   ]
}
```

8. Customer group price deleted for all websites, all groups, qty = 1 and fixed price.
```
{
   "events":[
      {
         "meta":{
            "type":"price_deleted",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":null,
            "cg":null
         }
      }
   ]
}
```

9. Customer group price added for the second website, retailer group, qty = 1 and discount percent price.
```
{
   "events":[
      {
         "meta":{
            "type":"price_changed",
            "price_type":"percent",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"3",
            "value":"15.00"
         }
      }
   ]
}
```

10. Customer group price deleted for the second website, retailer group, qty = 1 and discount percent price.
```
{
   "events":[
      {
         "meta":{
            "type":"price_deleted",
            "code":"price"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"3"
         }
      }
   ]
}
```

11. Customer group price added for the first website (base), wholesale groups, qty = 10 and fixed price.
```
{
   "events":[
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"fixed"
         },
         "data":{
            "id":"71",
            "w":"base",
            "cg":"2",
            "value":"5.000000",
            "qty":"10.0000"
         }
      }
   ]
}
```

12. Customer group price deleted for the first website (base), wholesale groups, qty = 10 and fixed price.
```
{
   "events":[
      {
         "meta":{
            "type":"tier_price_deleted"
         },
         "data":{
            "id":"71",
            "w":"base",
            "cg":"2",
            "qty":"10.0000"
         }
      }
   ]
}
```

13. Customer group price added for the second website, not logged in group, qty = 5 and discount percent price.
```
{
   "events":[
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"percent"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"0",
            "value":"5.00",
            "qty":"5.0000"
         }
      }
   ]
}
```

14. Customer group price deleted for the second website, not logged in group, qty = 5 and discount percent price.
```
{
   "events":[
      {
         "meta":{
            "type":"tier_price_deleted"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"0",
            "qty":"5.0000"
         }
      }
   ]
}
```

15. Multiple customer group prices added.
```
{
   "events":[
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"fixed"
         },
         "data":{
            "id":"71",
            "w":"base",
            "cg":"0",
            "value":"10.000000",
            "qty":"5.0000"
         }
      },
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"fixed"
         },
         "data":{
            "id":"71",
            "w":"base",
            "cg":"1",
            "value":"6.000000",
            "qty":"5.0000"
         }
      },
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"fixed"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"1",
            "value":"5.000000",
            "qty":"10.0000"
         }
      },
      {
         "meta":{
            "type":"tier_price_changed",
            "price_type":"percent"
         },
         "data":{
            "id":"71",
            "w":"second_website",
            "cg":"3",
            "value":"10.00",
            "qty":"4.0000"
         }
      }
   ]
}
```

16. Product custom option price
```
{
   "events":[
      {
         "meta":{
            "type":"custom_option_price_changed",
            "price_type":"fixed"
         },
         "data":{
            "id":"686", // TODO format id in correct way
            "w":null,
            "cg":null,
            "value":"13.000000"
         }
      }
   ]
}
```

17. Product custom option deleted
`TODO implement, trigger not working`

18. Downloadable link price add
```
{
   "events":[
      {
         "meta":{
            "type":"downloadable_link_price_changed"
         },
         "data":{
            "id":"3", // TODO format id in correct way
            "w":null,
            "cg":null,
            "value":"12.000000"
         }
      }
   ]
}
```

19. Downloadable link price remove
`TODO implement, trigger not working`

20. Configurable variation add
```
{
   "events":[
      {
         "meta":{
            "type":"variation_changed",
            "price_type":"configurable"
         },
         "data":{
            "id":"248",
            "w":null,
            "cg":null,
            "variation_id":"80"
         }
      }
   ]
}
```

21. Configurable variation remove
```
{
   "events":[
      {
         "meta":{
            "type":"variation_deleted",
            "price_type":"configurable"
         },
         "data":{
            "id":"248",
            "w":null,
            "cg":null,
            "variation_id":"80"
         }
      }
   ]
}
```

22. Bundle variation add
```
{
   "events":[
      {
         "meta":{
            "type":"variation_changed",
            "price_type":"bundle"
         },
         "data":{
            "id":"254",
            "w":null,
            "cg":null,
            "variation_id":"77"
         }
      }
   ]
}
```

23. Bundle variation remove
```
{
   "events":[
      {
         "meta":{
            "type":"variation_deleted",
            "price_type":"bundle"
         },
         "data":{
            "id":"254",
            "w":null,
            "cg":null,
            "variation_id":"77"
         }
      }
   ]
}
```

**Comment by bricht** [comment]
**Updated version**

     1. Product regular price changed in global scope
     2. Product special price changed in global scope (TODO special price from / to)


 ```
 {
    "events":[
       {
          "meta": {
            "type": "price_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
          {
             "id":"71",
             "value":"12.000000",
             "code":"price"
          },
          {
             "id":"71",
             "value":"10.000000",
             "code":"special_price"
          },

          ]
       }
    ]
 }
 ```


     1. Product regular price changed on product store view level (website scope of the attribute)
     2. Product special price changed on product store view level (website scope of the attribute) (TODO special price from / to)


 ```
{
    "events":[
       {
          "meta": {
            "type": "price_changed",
            "website": second_website,
            "customer_group": null
          },
          "data":[
          {
             "id":"71",
             "value":"15.000000",
             "code":"price"
          },
          {
             "id":"71",
             "value":"12.000000",
             "code":"special_price"
          }
          ]
       }
    ]
 }
 ```


     1. Special price deleted in global scope


 ```
{
    "events":[
       {
          "meta": {
            "type": "price_deleted",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"71",
                 "value":"null",
                 "code":"special_price"
              }
          ]
       }
    ]
 }

 ```

     1. Special price deleted on product store view level (website scope of the attribute)


 ```
{
    "events":[
       {
          "meta": {
            "type": "price_deleted",
            "website": "second_website",
            "customer_group": null
          },
          "data":[
              {
                 "id":"71",
                 "value":"null",
                 "code":"special_price"
              }
          ]
       }
    ]
 }

 ```

     1. Customer group price added for all websites, all groups, qty = 1 and fixed price.


 ```
{
    "events":[
       {
          "meta": {
            "type": "price_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"71",
                 "value":"8.000000",
                 "code":"price",
                 "price_type":"fixed"
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price deleted for all websites, all groups, qty = 1 and fixed price.


```
{
    "events":[
       {
          "meta": {
            "type": "price_deleted",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"71",
                 "value":null,
                 "code":"price",
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price added for the second website, retailer group, qty = 1 and discount percent price.

```
{
    "events":[
       {
          "meta": {
            "type": "price_changed",
            "website": second_website,
            "customer_group": "3"
          },
          "data":[
              {
                 "id":"71",
                 "value":"15.00",
                 "code":"price",
                 "price_type":"percent"
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price deleted for the second website, retailer group, qty = 1 and discount percent price.

```
{
    "events":[
       {
          "meta": {
            "type": "price_deleted",
            "website": "second_website",
            "customer_group": "3"
          },
          "data":[
              {
                 "id":"71",
                 "value":null,
                 "code":"price",
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price added for the first website (base), wholesale groups, qty = 10 and fixed price.


```
{
    "events":[
       {
          "meta": {
            "type": "tier_price_changed",
            "website": "base",
            "customer_group": "2"
          },
          "data":[
              {
                 "id":"71",
                 "value":"5.000000",
                 "price_type":"fixed",
                 "qty":"10.0000"
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price deleted for the first website (base), wholesale groups, qty = 10 and fixed price.

```
{
    "events":[
       {
          "meta": {
            "type": "tier_price_deleted",
            "website": "base",
            "customer_group": "2"
          },
          "data":[
              {
                 "id":"71",
                 "value":null,
                 "qty":"10.0000"
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price added for the second website, not logged in group, qty = 5 and discount percent price.

```
{
    "events":[
       {
          "meta": {
            "type": "tier_price_changed",
            "website": "second_website",
            "customer_group": "0"
          },
          "data":[
              {
                 "id":"71",
                 "value":"5.00",
                 "price_type":"percent",
                 "qty":"5.0000"
              }
          ]
       }
    ]
 }
 ```

     1. Customer group price deleted for the second website, not logged in group, qty = 5 and discount percent price.

```
{
    "events":[
       {
          "meta": {
            "type": "tier_price_deleted",
            "website": "second_website",
            "customer_group": "0"
          },
          "data":[
              {
                 "id":"71",
                 "value":null,
                 "qty":"5.0000"
              }
          ]
       }
    ]
 }
 ```

     1. Multiple customer group prices added.

```
{
    "events":[
       {
          "meta": {
            "type": "tier_price_changed",
            "website": "base",
            "customer_group": "0"
          },
          "data":[
              {
                 "id":"71",
                 "value":"10.00000",
                 "price_type":"fixed",
                 "qty":"5.0000"
              }
          ]
       },
       "events":[
       {
          "meta": {
            "type": "tier_price_changed",
            "website": "base",
            "customer_group": "1"
          },
          "data":[
              {
                 "id":"71",
                 "value":"6.00000",
                 "price_type":"fixed",
                 "qty":"5.0000"
              },
              {
                 "id":"71",
                 "value":"5.00000",
                 "price_type":"fixed",
                 "qty":"10.0000"
              }
          ]
       },
       {
          "meta": {
            "type": "second_website",
            "website": "base",
            "customer_group": "1"
          },
          "data":[
              {
                 "id":"71",
                 "value":"7.00000",
                 "price_type":"fixed",
                 "qty":"5.0000"
              }
          ]
       },
       {
          "meta": {
            "type": "second_website",
            "website": "base",
            "customer_group": "3"
          },
          "data":[
              {
                 "id":"71",
                 "value":"4.0000",
                 "price_type":"percent",
                 "qty":"4.0000"
              }
          ]
       }
    ]
 }
 ```

     1. Product custom option price

```
{
    "events":[
       {
          "meta": {
            "type": "custom_option_price_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"686", // TODO format id in correct way
                 "value":"13.000000",
                 "price_type":"fixed",
              }
          ]
       }
    ]
 }
 ```

     1. Product custom option deleted
        `TODO implement, trigger not working`

     2. Downloadable link price add


 ```
{
    "events":[
       {
          "meta": {
            "type": "downloadable_link_price_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"3", // TODO format id in correct way
                 "value":"12.000000"
              }
          ]
       }
    ]
}
 ```

     1. Downloadable link price remove
        `TODO implement, trigger not working`

     2. Configurable variation add


 ```
{
    "events":[
       {
          "meta": {
            "type": "variation_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"248",
                 "value":"12.000000",
                 "price_type":"configurable",
                 "variation_id":"80"
              }
          ]
       }
    ]
}
 ```

     1. Configurable variation remove


 ```

{
    "events":[
       {
          "meta": {
            "type": "variation_deleted",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":""248",
                 "value": null,
                 "price_type":"configurable",
                 "variation_id":"80"
              }
          ]
       }
    ]
}
 ```

     1. Bundle variation add


 ```
{
    "events":[
       {
          "meta": {
            "type": "variation_changed",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":"254",
                 "value":"12.000000",
                 "price_type":"bundle",
                 "variation_id":"77"
              }
          ]
       }
    ]
}
 ```

     1. Bundle variation remove


 ```

{
    "events":[
       {
          "meta": {
            "type": "variation_deleted",
            "website": null,
            "customer_group": null
          },
          "data":[
              {
                 "id":""254",
                 "value": null,
                 "price_type":"bundle",
                 "variation_id":"77"
              }
          ]
       }
    ]
}
 ```

**Comment by aleksejsbaranovs10** [comment]
**1. Product regular price changed in global scope
2. Product special price changed in global scope**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"price",
               "value":"54.000000"
            },
            {
               "id":"164",
               "attribute_code":"special_price",
               "value":"42.000000"
            }
         ]
      }
   ]
}
```

**3. Product regular price changed on product store view level (website scope of the attribute)
4. Product special price changed on product store view level (website scope of the attribute)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_changed",
            "website":"second_website",
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"price",
               "value":"51.000000"
            },
            {
               "id":"164",
               "attribute_code":"special_price",
               "value":"41.000000"
            }
         ]
      }
   ]
}
```

**5. Special price deleted in global scope**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"special_price",
               "value":null
            }
         ]
      }
   ]
}
```

**6. Special price deleted on product store view level (website scope of the attribute)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_deleted",
            "website":"second_website",
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"special_price",
               "value":null
            }
         ]
      }
   ]
}
```

**7. Customer group price added for all websites, all groups, qty = 1 and fixed price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"1.0000",
               "price_type":"fixed",
               "value":"33.000000"
            }
         ]
      }
   ]
}
```

**8. Customer group price deleted for all websites, all groups, qty = 1 and fixed price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"1.0000",
               "price_type":null,
               "value":null
            }
         ]
      }
   ]
}
```

**9. Customer group price added for the second website, retailer group, qty = 1 and discount percent price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_changed",
            "website":"second_website",
            "customer_group":"3"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"1.0000",
               "price_type":"percent",
               "value":"15.00"
            }
         ]
      }
   ]
}
```

**10. Customer group price deleted for the second website, retailer group, qty = 1 and discount percent price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"price_deleted",
            "website":"second_website",
            "customer_group":"3"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"1.0000",
               "price_type":null,
               "value":null
            }
         ]
      }
   ]
}
```

**11. Customer group price added for the first website (base), wholesale groups, qty = 10 and fixed price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"tier_price_changed",
            "website":"base",
            "customer_group":"2"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"10.0000",
               "price_type":"fixed",
               "value":"5.000000"
            }
         ]
      }
   ]
}
```

**12. Customer group price deleted for the first website (base), wholesale groups, qty = 10 and fixed price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"tier_price_deleted",
            "website":"base",
            "customer_group":"2"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"10.0000",
               "price_type":null,
               "value":null
            }
         ]
      }
   ]
}
```

**13. Customer group price added for the second website, not logged in group, qty = 5 and discount percent price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"tier_price_changed",
            "website":"second_website",
            "customer_group":"0"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"5.0000",
               "price_type":"percent",
               "value":"10.00"
            }
         ]
      }
   ]
}
```

**14. Customer group price deleted for the second website, not logged in group, qty = 5 and discount percent price.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"tier_price_deleted",
            "website":"second_website",
            "customer_group":"0"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"5.0000",
               "price_type":null,
               "value":null
            }
         ]
      }
   ]
}
```

**15. Multiple customer group prices added.**
```
{
   "events":[
      {
         "meta":{
            "event_type":"tier_price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"22.0000",
               "price_type":"fixed",
               "value":"11.000000"
            }
         ]
      },
      {
         "meta":{
            "event_type":"tier_price_changed",
            "website":"base",
            "customer_group":"1"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"10.0000",
               "price_type":"fixed",
               "value":"7.000000"
            },
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"25.0000",
               "price_type":"fixed",
               "value":"5.000000"
            }
         ]
      },
      {
         "meta":{
            "event_type":"tier_price_changed",
            "website":"second_website",
            "customer_group":"2"
         },
         "data":[
            {
               "id":"164",
               "attribute_code":"tier_price",
               "qty":"11.0000",
               "price_type":"fixed",
               "value":"23.000000"
            }
         ]
      }
   ]
}
```

**16. Product custom option price (entered option)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"custom_option_price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjY=",
               "value":"35.000000",
               "price_type":"fixed"
            }
         ]
      }
   ]
}
```

**17. Product custom option deleted (entered option)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"custom_option_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjY="
            }
         ]
      }
   ]
}
```

**18. Product custom option price (selectable option)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"custom_option_type_price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjcvNzI3",
               "value":"35.000000",
               "price_type":"fixed"
            },
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjcvNzI4",
               "value":"40.000000",
               "price_type":"fixed"
            },
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjcvNzI5",
               "value":"45.000000",
               "price_type":"fixed"
            }
         ]
      }
   ]
}
```

**19. Product custom option deleted (selectable option)**
```
{
   "events":[
      {
         "meta":{
            "event_type":"custom_option_type_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjcvNzI4"
            },
            {
               "id":"Y3VzdG9tLW9wdGlvbi83MjcvNzI5"
            }
         ]
      }
   ]
}
```

**20. Downloadable link price add**
```
{
   "events":[
      {
         "meta":{
            "event_type":"downloadable_link_price_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"ZG93bmxvYWRhYmxlLzMz",
               "value":"55.000000"
            }
         ]
      }
   ]
}
```

**21. Downloadable link price remove**
```
{
   "events":[
      {
         "meta":{
            "event_type":"downloadable_link_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"ZG93bmxvYWRhYmxlLzMz"
            }
         ]
      }
   ]
}
```

**22. Configurable variation add**
```
{
   "events":[
      {
         "meta":{
            "event_type":"variation_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"171",
               "variation_id":"169",
               "price_type":"configurable"
            },
            {
               "id":"171",
               "variation_id":"170",
               "price_type":"configurable"
            }
         ]
      }
   ]
}
```

**23. Configurable variation remove**
```
{
   "events":[
      {
         "meta":{
            "event_type":"variation_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"171",
               "variation_id":"169",
               "price_type":"configurable"
            },
            {
               "id":"171",
               "variation_id":"170",
               "price_type":"configurable"
            }
         ]
      }
   ]
}
```

**24. Bundle variation add**
```
{
   "events":[
      {
         "meta":{
            "event_type":"variation_changed",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"176",
               "variation_id":"169",
               "price_type":"bundle"
            }
         ]
      }
   ]
}
```

**25. Bundle variation remove**
```
{
   "events":[
      {
         "meta":{
            "event_type":"variation_deleted",
            "website":null,
            "customer_group":null
         },
         "data":[
            {
               "id":"176",
               "variation_id":"169",
               "price_type":"bundle"
            }
         ]
      }
   ]
}
```
